azurerm/data_source_nat_gateway.go

```diff
@@ -0,0 +1,134 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNatGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"idle_timeout_in_minutes": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"public_ip_address_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"public_ip_prefix_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"resource_guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
Can we update this attribute name to be `sku_name` instead of `sku`? We are deprecating the use of `sku` in Terraform as a whole and flattening the structure out into a single string called `sku_name`.
```suggestion
			},
			
			"sku_name": { 
				Type:     schema.TypeString, 
				Computed: true, 
			},
```

Reply:
Done